### PR TITLE
feat: added membership creation API

### PR DESCRIPTION
**Pull request by scmmishra**
This PR has the following changes
1. Set member email as unique
1. Created API for creation of membership

Version 12 PR: https://github.com/frappe/erpnext/pull/20944

### CODE REVIEW

File: erpnext/non_profit/doctype/membership/membership.py [hunk 1]
```diff
@@ -50,4 +50,33 @@ def on_payment_authorized(self, status_changed_to=None):
...
+	member_list = frappe.get_all("Member", filters={'email': email})
+	if len(member_list):
+		return member_list[0]['name']
+	else:
+		new_member = frappe.new_doc('Member')
+		new_member.member_name = name
+		new_member.membership_type = membership_type
+		new_member.email = email
+		new_member.insert()
+		new_member.load_from_db()
```
Comment (nabinhait) [inline erpnext/non_profit/doctype/membership/membership.py]: why load_from_db?

File: erpnext/non_profit/doctype/membership/membership.py [hunk 2]
```diff
@@ -50,4 +50,33 @@ def on_payment_authorized(self, status_changed_to=None):
 			self.load_from_db()
 			self.db_set('paid', 1)
 
+@frappe.whitelist()
+def create_membership(name, email, membership_type, from_date, to_date, currency, amount, paid):
```
Comment (nabinhait) [inline erpnext/non_profit/doctype/membership/membership.py]: So many mandatory args?

File: erpnext/non_profit/doctype/membership/membership.py (hunk 1 above)
Comment (scmmishra) [reply erpnext/non_profit/doctype/membership/membership.py]: Removed this

File: erpnext/non_profit/doctype/membership/membership.py (hunk 2 above)
Comment (scmmishra) [reply erpnext/non_profit/doctype/membership/membership.py]: Set a few to defaults, but most of these args are required

**Pull request closed without merging**